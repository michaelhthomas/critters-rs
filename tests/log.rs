use std::{fs, path::PathBuf};

use critters_rs::{Critters, CrittersOptions};
use insta::assert_snapshot;

#[test]
fn skip_invalid_path() {
    mock_logger::init();

    let path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("tests/src/subpath");

    let critters = Critters::new(CrittersOptions {
        reduce_inline_styles: false,
        external: true,
        path: path.to_string_lossy().to_string(),
        ..Default::default()
    });

    let html = fs::read_to_string(path.join("subpath-validation.html")).unwrap();
    let result = critters.process(&html).expect("Failed to process html.");

    mock_logger::MockLogger::entries(|entries| {
        assert!(
            entries.iter().any(|l| l.level == log::Level::Warn
                && l.body.contains("not within the configured output path")
                && l.body.contains("styles.css")),
            "{}",
            entries
                .iter()
                .filter(|l| l.body.contains("critters_rs"))
                .map(|l| l.body.clone())
                .collect::<Vec<_>>()
                .join("\n")
        );
    });
    assert_snapshot!(result);
}
