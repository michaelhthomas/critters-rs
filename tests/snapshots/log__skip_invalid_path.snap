---
source: tests/log.rs
expression: result
---
<html><head><link>
    <title>Testing</title>
    <link rel="stylesheet" href="../styles.css">
    <style>body{height:100%}</style><link rel="preload" href="./styles2.css" as="style">
  </head>
  <body>
    <div class="container">
      <header>
        <div class="banner">Lorem ipsum dolor sit amet</div>
      </header>
    </div>
    <div class="" data-critters-container="">
      <h1>Hello World!</h1>
      <p>This is a paragraph</p>
      <input class="input-field">
      <div class="contents"></div>
      <div class="selected active"></div>
    </div>
    <footer class=""></footer>
  

<link rel="stylesheet" href="./styles2.css"></body></html>
